npm_and_yarn/lib/dependabot/npm_and_yarn/file_updater/npm_lockfile_updater.rb

```diff
@@ -256,6 +236,41 @@ def flattenend_manifest_dependencies
             end
         end
 
+        # Runs `npm install` with `--package-lock-only` flag to update the
+        # lockfiile.
+        #
+        # Other npm flags:
+        # - `--force` ignores checks for platform (os, cpu) and engines
+        # - `--dry-run=false` the updater sets a global .npmrc with dry-run:
+        #   true to work around an issue in npm 6, we don't want that here
+        # - `--ignore-scripts` disables prepare and prepack scripts which are
+        #   run when installing git dependencies
+        def run_npm_install_lockfile_only(*install_args)
+          command = [
+            "npm",
+            "install",
+            *install_args,
+            "--force",
+            "--dry-run",
+            "false",
+            "--ignore-scripts",
+            "--package-lock-only"
+          ].join(" ")
+
+          fingerprint = [
+            "npm",
+            "install",
+            install_args.empty? ? "" : "<install_args>",
+            "--force",
+            "--dry-run",
+            "false",
+            "--ignore-scripts",
+            "--package-lock-only"
+          ].join(" ")
```

Comment:
I realize this is copy/pasted, but doesn't seem a little weird this is copy/pasted vs an inline function (does Ruby allow those?)  where the fingerprint feeds in a different value than the command call...  nbd though.

Reply:
Yeah, I dunno if this is doable in Ruby, but in python or Go I'd do something like:

```
        def run_npm_install_lockfile_only(*install_args)
           def generate_command_string(install_args)      
            [
              "npm",
              "install",
              install_args,
              "--force",
              "--dry-run",
              "false",
              "--ignore-scripts",
              "--package-lock-only"
            ].join(" ")

          command = generate_command_string((*install_args).join(" "))
          # Make `install_args` generic to Sentry for easier grouping and avoid risk of exposing PII
          fingerprint = generate_command_string((install_args.empty? ? "" : "<install_args>"))
```

No idea if it's idiomatic in Ruby though...

Reply:
Ahhh, I see. You'd use a lambda for that, see https://rubyapi.org/3.1/o/proc. Agree on doing it separately since it should be happening everywhere and would deviate too much from the point of this PR.

Reply:
![Screenshot_20241216-142444.jpg](https://github.com/user-attachments/assets/ae593268-3dcd-45ac-89cf-84d51bff6a11)